src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FailOverSupportedConfigurationClient.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class FailOverSupportedConfigurationClient : IConfigurationClient
+    {
+        private const int HttpStatusRequestThrottled = 429;
+
+        private readonly IEnumerable<LocalConfigurationClient> clients;
+
+        private DateTimeOffset retryPrimaryConfigClientAfter;
+
+        private int failedRequestsToPrimaryConfigClient = 0;
+
+        public FailOverSupportedConfigurationClient(IEnumerable<LocalConfigurationClient> clients)
+        {
+            if (clients == null || clients.Count() < 1)
+            {
+                throw new ArgumentNullException(nameof(clients));
+            }
+
+            this.retryPrimaryConfigClientAfter = DateTimeOffset.UtcNow;
+            this.failedRequestsToPrimaryConfigClient = 0;
+            this.clients = clients;
+        }
+
+        public Task<Response<ConfigurationSetting>> GetConfigurationSettingAsync(string key, string label = null, CancellationToken cancellationToken = default)
+        {
+            return ExecuteWithFailOverPolicyAsync(clients.Select((client) => new Func<Task<Response<ConfigurationSetting>>>(() => client.Client.GetConfigurationSettingAsync(key, label, cancellationToken))), cancellationToken);
```

Comment:
Rather than creating all the delegates each time, does it worth with the pattern

```
    Execute(
        clients,
        (client) => client.DoSomething(arg1, cancellationToken),
        cancellationToken);
```
?

Since `Clients` is a member, no need to pass it the the method even.

Reply:
With this change, client object is available in the ExecuteWithPolicy code. That way we can associate backoff properties with each client if necessary.